lesson01.calculator/app.js

```diff
@@ -0,0 +1,387 @@
+});
+
+document.querySelector('.two').addEventListener('click', function(){
+    numberClick('.two');  
+});
+
+document.querySelector('.three').addEventListener('click', function(){
+    numberClick('.three');
+});
+
+document.querySelector('.four').addEventListener('click', function(){
+    numberClick('.four');
+});
+
+document.querySelector('.five').addEventListener('click', function(){
+    numberClick('.five');
+});
+
+document.querySelector('.six').addEventListener('click', function(){
+    numberClick('.six');
+});
+
+document.querySelector('.seven').addEventListener('click', function(){
+    numberClick('.seven');   
+});
+
+document.querySelector('.eight').addEventListener('click', function(){
+    numberClick('.eight');
+});
+
+document.querySelector('.nine').addEventListener('click', function(){
+    numberClick('.nine');
+});
+
+
+
+
+//----------------Event Listeners for all the math operations buttons----------//
+document.querySelector('.plus').addEventListener('click', function(){
+    operatorClick('.plus', '+');
+    
+});
+
+document.querySelector('.div').addEventListener('click', function(){
+    operatorClick('.div', '/'); 
+});
+
+document.querySelector('.multi').addEventListener('click', function(){
+    operatorClick('.multi', '*');
+});
+
+document.querySelector('.minus').addEventListener('click', function(){
+    operatorClick('.minus', '-'); 
+});
+
+
+document.querySelector('.equal').addEventListener('click', function(){
+    
+    screenCheck();
+    
+    if(screen.length === 0)
+        return;
+    
+    if(screen[screen.length-1] === '.')
+        return;
+    
+    
+    //taking care of the case when clicking on equal and only minus and a number is on the screen (e.g, "- <NUM>")
+    if(elementsHistory[elementsHistory.length-1] === '-' && screen.length > 1){
+        elementsHistory[elementsHistory.length-2] = parseFloat(elementsHistory[elementsHistory.length-2]) * (-1);
+        elementsHistory.pop();
+    }
+    
+    
+    //taking care of the case when clicking equal but the last action was an operator
+    for (let i = 0 ; i <= operations.length-1 ; i++){
+        if(screen[screen.length-1] === operations[i])
+            return;
+    }
+    
+    
+    //fixing the elementsHistory array when there are negative numbers in the arithmetic expression
+    for (let i = elementsHistory.length-1 ; i >= 0 ; i--){
+        for (let j = 0 ; j <= operations.length-1 ; j++){
+            if(elementsHistory[i] === operations[j] && elementsHistory[i-1] === '-'){
+                elementsHistory[i-2] = parseFloat(elementsHistory[i-2]) * (-1);
+                elementsHistory.splice(i-1,1);
+            }
+        }
+    }
+    
+    
+    for (let i = 0; i <= elementsHistory.length-1  ; i+=2) {
+        elementsHistory[i] = parseFloat(elementsHistory[i]);
+    }
+    
+    
+    //calculation the arithmetic expression
+    let result;
+    
+    for (let i = elementsHistory.length - 1; i >= 0 ; i--) {
```

Comment:
This is a complex logic.
Try to thing about simple, more readable logic
Maybe you shouldn't allow operations with more than one operator, like : `1+2+3` - you can return `3` after `1+2` and let the user continue from there.